mlir/include/mlir/Dialect/LLVMIR/LLVMAttrDefs.td

```diff
@@ -1104,4 +1104,23 @@ def TailCallKindAttr : LLVM_Attr<"TailCallKind", "tailcallkind"> {
   let assemblyFormat = "`<` $tailCallKind `>`";
 }
 
+//===----------------------------------------------------------------------===//
+// WorkgroupAttribAttr
+//===----------------------------------------------------------------------===//
+
+def WorkgroupAttribAttr
+    : LLVM_Attr<"WorkgroupAttrib", "mlir.workgroup_attrib"> {
+  let summary = "GPU workgroup attribution information";
+  let description = [{
+    GPU workgroup attributions are `gpu.func` arguments encoding memory
```

Comment:
The doc sounds a bit like the workgroup attributions are argument value + this attribute? Aren't the attributions just the argument attributes attached to the pointer to preserve size and type of the allocation?

Reply:
I think it may be clearer now